lib/mumukit/templates/multi_file_precompile_hook.rb

```diff
@@ -0,0 +1,30 @@
+module Mumukit
+  class Templates::MultiFilePrecompileHook < Mumukit::Hook
+    include Mumukit::Templates::WithMultipleFiles
+
+    def compile(request)
+      files = files_of request
+      content = files.empty? ? single_file_content(request) : multi_file_content(files)
+
+      struct request.to_h.merge content: content
+    end
+
+    def main_file
+      raise NotImplementedError
+    end
+
+    def consolidate(main_content, files)
```

Comment:
You can use `required :main_file`

Reply:
But then it won't be so obvious that `consolidate` must receive `main_content` and `files`, right? I find that important since we don't have documentation for template hooks